### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by dwarring** [p6-xml/LibXML-p6#21]
Memory leak on failing XML parser. Test adapted from t/11memory.t
```
use LibXML;
for ^10000 {
    try {
        LibXML.parse: :string("<X>foo</Y>");
    }
}
```

I notice the DESTROY method never gets called on the created LibXML::Parser::Context object.

`$ perl6 -v
This is Rakudo version 2019.07.1-306-gfce964231 built on MoarVM version 2019.07.1-180-gcf3b545e5`

**Comment by dwarring** [comment]
Err, it's even more basic. DESTROY method isn't being called on LibXML::Parser::Context for
```
use LibXML;
for ^10000 {
    LibXML.parse: :string("<X>foo</X>");
}
```

**Comment by dwarring** [comment]
GC works for the simple case, of creating these objects directly:
```
use LibXML::Parser::Context;
for ^10000 {
     LibXML::Parser::Context.new;
}
```
Also worth noting that output LibXML::Document objects (which keep a reference to the context) are being destroyed.

**Comment by dwarring** [comment]
This is enough to stop GC:
```
use LibXML::Native;
use LibXML::Parser::Context;
for ^10000 {
    LibXML::Parser::Context.new.set-native: xmlParserCtxt.new;
}
```

**Comment by dwarring** [comment]
Further golfed to the setting of the error handling callback closure. Have raised Rakudo ticket https://github.com/rakudo/rakudo/issues/3184

**Comment by dwarring** [comment]
Fixed in 0.0.8. Switched from closures to dynamic variables ($*XML-CONTEXT). LibXML::Parser::Context objects are being destroyed. 11memory.t is running leaner.
